pkg/operator/controller/externaldns/pod.go

```diff
@@ -325,30 +326,23 @@ func (b *externalDNSContainerBuilder) fillAWSFields(container *corev1.Container)
 
 	env := []corev1.EnvVar{
 		{
-			Name: awsAccessKeyIDEnvVar,
-			ValueFrom: &corev1.EnvVarSource{
-				SecretKeyRef: &corev1.SecretKeySelector{
-					LocalObjectReference: corev1.LocalObjectReference{
-						Name: b.secretName,
-					},
-					Key: awsAccessKeyIDKey,
-				},
-			},
-		},
-		{
-			Name: awsAccessKeySecretEnvVar,
-			ValueFrom: &corev1.EnvVarSource{
-				SecretKeyRef: &corev1.SecretKeySelector{
-					LocalObjectReference: corev1.LocalObjectReference{
-						Name: b.secretName,
-					},
-					Key: awsAccessKeySecretKey,
-				},
-			},
+			Name:  awsCredentialEnvVarName,
+			Value: awsCredentialsPath,
 		},
```

Comment:
Another environment variable called `AWS_SDK_LOAD_CONFIG` with value "1".
[Reference](https://docs.aws.amazon.com/sdk-for-go/api/aws/session/).

Reply:
ExternalDNS' upstream uses [the enabled shared config option](https://github.com/kubernetes-sigs/external-dns/blob/f3e04dfe8c46853966b878ec76f3eeb6b2ed1441/provider/aws/aws.go#L194) which doesn't need this variable:

> Using the NewSessionWithOptions with SharedConfigState set to SharedConfigEnable will create the session as if the AWS_SDK_LOAD_CONFIG environment variable was set.